src/libponyrt/sched/scheduler.c

```diff
@@ -491,3 +492,71 @@ PONY_API pony_ctx_t* pony_ctx()
   pony_assert(this_scheduler != NULL);
   return &this_scheduler->ctx;
 }
+
+void ponyint_sched_mute(pony_ctx_t* ctx, pony_actor_t* sender, pony_actor_t* recv)
+{
+  scheduler_t* sched = ctx->scheduler;
+  size_t index = HASHMAP_UNKNOWN;
+  muteref_t key;
+  key.key = recv;
+
+  muteref_t* mref = ponyint_mutemap_get(&sched->mute_mapping, &key, &index);
+  if(mref == NULL)
+  {
+    mref = ponyint_muteref_alloc(recv);
+    ponyint_mutemap_putindex(&sched->mute_mapping, mref, index);
+  }
+
+  size_t index2 = HASHMAP_UNKNOWN;
+  pony_actor_t* r = ponyint_muteset_get(&mref->value, sender, &index2);
+  if(r == NULL)
+  {
+    ponyint_muteset_putindex(&mref->value, sender, index2);
+    uint8_t muted = atomic_load_explicit(&sender->muted, memory_order_relaxed);
+    muted++;
+    atomic_store_explicit(&sender->muted, muted, memory_order_relaxed);
+  }
+}
+
+void ponyint_sched_unmute_senders(pony_ctx_t* ctx, pony_actor_t* actor, bool inform)
+{
+  scheduler_t* sched = ctx->scheduler;
+  size_t index;
+  muteref_t key;
+  key.key = actor;
+
+  if(inform)
+  {
+    for(uint32_t i = 0; i < scheduler_count; i++)
+    {
+      if(&scheduler[i] != sched)
+        send_msg(i, SCHED_UNMUTE_ACTOR, (intptr_t)actor);
+    }
+  }
+
+  muteref_t* mref = ponyint_mutemap_get(&sched->mute_mapping, &key, &index);
+
+  if(mref != NULL)
+  {
+    size_t i = HASHMAP_UNKNOWN;
+    pony_actor_t* muted = NULL;
+
+    while((muted = ponyint_muteset_next(&mref->value, &i)) != NULL)
+    {
+      uint8_t muted_count = atomic_load_explicit(&muted->muted, memory_order_relaxed);
+      pony_assert(muted_count > 0);
+      muted_count--;
+      atomic_store_explicit(&muted->muted, muted_count, memory_order_relaxed);
+
+      if (muted->muted == 0)
```

Comment:
This needs to either be an `atomic_load_explicit`, or use the `muted_count` local.

Reply:
thanks i missed that